tests/testthat/testEdgeThresh.R

```diff
@@ -0,0 +1,251 @@
+    expect_error(.permuteGraph(graph = myEGG, Q = -1))
+
+    expect_error(.permuteGraph(graph = 10,    Q = 100))
+
+}
+# ((Should valid inputs different from rete.EGGprototype be used?))
+# ((The testing below can get quite computationally expensive...))
+test_that("a sane input to .permuteGraph gives an expected output", {
+
+    ### Test for Q = 1, 10, 20, 50, 100, 200
+    myPermutation <- .permuteGraph(myEGG, Q = 1)
+
+    # Check for self-loops and multiple edges
+    expect_true(igraph::is.simple(myPermutation))
+
+    # Check that the size of the largest connected component has not changed
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+
+    # Check that overall size has not changed
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 10)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 20)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 50)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 100)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 200)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+})
+# ((Is it possible to pass permuted graphs from the previous test section
+# to the next one, if they pass? This would minimize a lot of computation))
+test_that("a sane input to edgeThresh gives an expected output", {
+    # Provide small input data, provide small output data.
+    #   (Note that the output needs to be independently verifiably correct.
+    #    Don't run an erroneous function, and then test against the erroneous
+    #    output you received when you ran your function for the first time.)
+    # Try this with important variations of parameters (reasonable, not
+    #    combinatorially exhaustive).
+    # Cover your code.
+
+    # ((Is graph.attributes(gWithThresh)$delta the correct way of retrieving
+    # delta values from graph attributes?)))
+    ### Check edgeThresh with different valid input values
+    ### For each check, length deltaVector should be Lmax - 1, numeric, and
+    ### between 0 and 1.
```

Comment:
@hyginn Sorry I meant the value for each delta should be between 0 and 1. Isn't this true?